compiler/src/test/scala/org/scalajs/core/compiler/test/ScalaJSDefinedTest.scala

```diff
@@ -400,6 +472,26 @@ class ScalaJSDefinedTest extends DirectTest with TestHelpers {
       |      @JSName(A.a)
       |       ^
     """
+
+    """
+    object A {
+      val a = "Hello"
+      final val b = "World"
+    }
+
+    @ScalaJSDefined
+    object B extends js.Object {
+      @JSName(A.a)
+      def foo: Int = js.native
+      @JSName(A.b)
+      def bar: Int = js.native
+    }
+    """ hasErrors
+    """
+      |newSource1.scala:12: error: The argument to JSName must be a literal string
+      |      @JSName(A.a)
+      |       ^
+    """
```

Comment:
These test cases are potentially unstable, since there are other errors in the class... (I don't know how the one for the class slipped my review)